data-provider/class-btm-task-dao.php

```diff
@@ -287,6 +287,80 @@ public function mark_as_in_progress( I_BTM_Task $task ){
 		return $this->update( $task );
 	}
 
+	/**
+	 * This function should be update only registered and in progress tasks to paused tasks
+	 *
+	 * @param array $ids
+	 *
+	 * @return bool
+	 */
+	public function mark_as_paused_many_by_ids( array $ids ){
+		global $wpdb;
+
+		if( 0 === count( $ids ) ){
+			return true;
+		}
+
+		$ids_in = '';
+		foreach ( $ids as $id ){
+			$ids_in .= $wpdb->prepare(', %d', $id);
+		}
+
+		$ids_in = ltrim( $ids_in, ', ' );
+
+		$updated = $wpdb->query( '
+			UPDATE `' . $this->get_table_name() . '`
+			SET `status` =  "'.BTM_Task_Run_Status::STATUS_PAUSED.'"
+			WHERE `id` IN ( ' . $ids_in . ' )
+			AND ( 
+					`status` = "'.BTM_Task_Run_Status::STATUS_REGISTERED.'" 
+				OR 
+					`status` = "'.BTM_Task_Run_Status::STATUS_IN_PROGRESS.'" 
+				)
+		' );
+
+		if( $updated ){
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * This function should be update only paused tasks to registered tasks
```

Comment:
@eriktad I'm not sure if function name and description must "force" to change status of paused tasks to 'registered', I guess function name must be like "resume_tasks" and then, in description, we can identify, that new status will be 'registered' after resume.

Reply:
Fixed